contribution_checker/_commits.py

```diff
@@ -11,40 +11,56 @@
 
 from git import Repo, exc
 
+from contribution_checker._helper import clone_or_pull_repository, get_cache_dir
 from contribution_checker._report import RepoReport
 
 
-def extract_matching_commits(report: RepoReport, repoinfo: tuple, pattern: str) -> list:
+def extract_matching_commits(report: RepoReport, repoinfo: dict, pattern: str) -> list:
     """Clone a repository and get all its commits"""
-    repopath, repotype = repoinfo
 
     # Remote repository, clone into temp directory
-    if repotype == "remote":
-        with tempfile.TemporaryDirectory() as tmpdir:
-            logging.info("Attempting to extract commits from remote repository")
-            logging.info("Cloning %s to %s", repopath, tmpdir)
-            repo = Repo.clone_from(url=repopath, to_path=tmpdir)
+    if repoinfo["remote"]:
+        logging.info("Using remote repository: %s", report.path)
 
-            all_commits = _extract_all_commits(report, repo)
+        # Define path the remote repository shall be cloned to
+        if repoinfo["cache"]:
+            repodir = get_cache_dir(report.path)
+        else:
+            repodir_object = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
```

Comment:
Why do we still need the option not to cache?

Reply:
If a user wants to check many and/or very large repositories, they may want to avoid storing a local copy of these repos that may use up a lot of disk space.

This makes me think about the temporary directory concept. When cloning a large repo, `/tmp` may fill up quite quickly. Perhaps this could be an expert setting.